modal/object.py

```diff
@@ -162,40 +162,42 @@ async def _lookup(
 
 
 class _Provider(Generic[H]):
+    _load: Callable[[Resolver, Optional[str]], Awaitable[H]]
+    _preload: Optional[Callable[[Resolver, Optional[str]], Awaitable[H]]]
+
     def _init(
         self,
-        load: Callable[[Resolver, str], Awaitable[H]],
+        load: Callable[[Resolver, Optional[str]], Awaitable[H]],
         rep: str,
         is_persisted_ref: bool = False,
-        preload: Optional[Callable[[Resolver, str], Awaitable[H]]] = None,
+        preload: Optional[Callable[[Resolver, Optional[str]], Awaitable[H]]] = None,
     ):
         self._local_uuid = str(uuid.uuid4())
         self._load = load
-        if preload is not None:
-            # only sets _preload if one is provided, otherwise a noop is used
-            self._preload = preload
+        self._preload = preload
         self._rep = rep
         self.is_persisted_ref = is_persisted_ref
 
     def __init__(
         self,
-        load: Optional[Callable[[Resolver, str], Awaitable[H]]],
+        load: Optional[Callable[[Resolver, Optional[str]], Awaitable[H]]],
```

Comment:
Nit: Just noticed load is also Optional here, which it probably shouldn't be (as of now). I think I might have committed it by mistake when I was experimenting with making load optional just like preload.

Reply:
yeah that's right, it shouldn't be optional – thanks!